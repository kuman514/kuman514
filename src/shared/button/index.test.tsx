import { fireEvent, render, screen } from '@testing-library/react';

import Button from '.';

describe('Button', () => {
  it('should render as expected', () => {
    const { container } = render(<Button onClick={() => {}}>Yasuo</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be clickable when it is not disabled', async () => {
    const mockFn = jest.fn();
    render(<Button onClick={mockFn}>Yasuo</Button>);
    fireEvent.click(await screen.findByText(/Yasuo/i));
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should not be clickable when it is disabled', async () => {
    const mockFn = jest.fn();
    render(
      <Button isDisabled onClick={mockFn}>
        Yasuo
      </Button>
    );
    fireEvent.click(await screen.findByText(/Yasuo/i));
    expect(mockFn).not.toHaveBeenCalled();
  });
});
