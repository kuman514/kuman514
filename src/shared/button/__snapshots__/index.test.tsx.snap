// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render as expected 1`] = `
<button
  class="bg-green-600 hover:bg-green-400 text-white px-6 py-4 rounded-lg transition hover:-translate-y-1 hover:drop-shadow-[0_4px_8px_rgba(0,0,0,0.5)] undefined"
>
  Yasuo
</button>
`;
